src/Generator/PropertyGenerator.php

```diff
@@ -136,40 +148,62 @@ public function __construct($name = null, $defaultValue = null, $flags = self::F
         }
     }
 
-    /**
-     * @param  bool $const
-     * @return PropertyGenerator
-     */
-    public function setConst($const)
+    public function setConst(bool $const): self
     {
-        if ($const) {
-            $this->setFlags(self::FLAG_CONSTANT);
-        } else {
-            $this->removeFlag(self::FLAG_CONSTANT);
+        if (true === $const) {
+            return $this->setFlags(self::FLAG_CONSTANT);
         }
 
-        return $this;
+        return $this->removeFlag(self::FLAG_CONSTANT);
+    }
+
+    public function isConst(): bool
+    {
+        return (bool) ($this->flags & self::FLAG_CONSTANT);
+    }
+
+    public function setReadonly(bool $readonly): self
+    {
+        if (true === $readonly) {
+            return $this->setFlags(self::FLAG_READONLY);
+        }
+
+        return $this->removeFlag(self::FLAG_READONLY);
+    }
+
+    public function isReadonly(): bool
```

Comment:
We should probably get rid of this public API - let's consider `#generate()` our main API, while reading state is probably not relevant (and not the use-case for this class)

Reply:
We use it in tests, also this version provides such an interface for all other modifiers, imo we should keep it. Also keep in mind this is just a bool, so not much which can be misused?! But your decision!

Reply:
Remember that any API we add is API we have to maintain long-term too.

No big issue here though, let's add it 👍